### PR TITLE
feat: adding updating profile flow

### PR DESCRIPTION
**Pull request by rasoul-shahriari**

### CODE REVIEW

File: src/modules/authenticated/authenticated.tsx [hunk 1]
```diff
@@ -42,8 +42,8 @@ const screens = [
   },
   {
     name: 'authenticated-settings',
-    component: AuthenticatedSettings,
-    title: 'Settings',
+    component: authenticatedSettings,
```
Comment (rasoul-shahriari) [inline src/modules/authenticated/authenticated.tsx]: authenticatedProfile is better I guess :)

Comment (Mokto) [inline src/modules/authenticated/authenticated.tsx]: PascalCase

File: src/modules/authenticated/settings/index.tsx [hunk 2]
```diff
@@ -1,31 +1,68 @@
...
+  },
+});
 
-export const AuthenticatedSettings = memo(() => {
+export const authenticatedSettings = memo(() => {
   const navigation = useNavigation();
+  const profileFlowStore = useProfileFlowStore();
   const userStore = useUserStore();
 
+  profileFlowStore.firstName = userStore.profile?.firstName;
```
Comment (Mokto) [inline src/modules/authenticated/settings/index.tsx]: useEffect

File: src/modules/authenticated/settings/index.tsx [hunk 3]
```diff
@@ -1,31 +1,68 @@
...
+          <CheckBoxButton
+            onPress={() => navigation.navigate('profile-infos-phone')}
+            checked={!!profileFlowStore.phone}
+          >
+            Phone
+          </CheckBoxButton>
+        </View>
+        <Button
+          theme="grey"
+          style={{ marginTop: 80 }}
```
Comment (Mokto) [inline src/modules/authenticated/settings/index.tsx]: use styles

File: src/modules/profile/infos/name.tsx [hunk 4]
```diff
@@ -30,6 +30,12 @@ const styles = StyleSheet.create({
 export const ProfileInfosName = observer(() => {
   const navigation = useNavigation();
   const profileFlowStore = useProfileFlowStore();
+  const userStore = useUserStore();
+
+  const [firstName, SetFirstName] = useState<string>(userStore?.profile?.firstName || '');
```
Comment (Mokto) [inline src/modules/profile/infos/name.tsx]: setFirstName

File: src/modules/profile/infos/name.tsx [hunk 5]
```diff
@@ -30,6 +30,12 @@ const styles = StyleSheet.create({
...
+  const [lastName, SetLastName] = useState<string>(
```
Comment (Mokto) [inline src/modules/profile/infos/name.tsx]: setLastName

File: src/modules/profile/infos/name.tsx [hunk 6]
```diff
@@ -41,24 +47,39 @@ export const ProfileInfosName = observer(() => {
...
         placeholder="Doe"
         // placeholderTextColor={colors.blue}
       />
 
       <View style={styles.buttonContainer}>
-        <Button onPress={navigation.goBack}>Done</Button>
+        <Button
+          onPress={async () => {
+            if (userStore.profile._id) {              
+              await userStore.patchProfile(userStore.profile._id, {
```
Comment (Mokto) [inline src/modules/profile/infos/name.tsx]: I don't see why we await we the end of the result. Let's do it in "optimistic update" mode

File: src/modules/profile/infos/name.tsx [hunk 7]
```diff
@@ -41,24 +47,39 @@ export const ProfileInfosName = observer(() => {
...
+        onChangeText={lastName => SetLastName(lastName)}
         autoCorrect={false}
         placeholder="Doe"
         // placeholderTextColor={colors.blue}
       />
 
       <View style={styles.buttonContainer}>
-        <Button onPress={navigation.goBack}>Done</Button>
+        <Button
+          onPress={async () => {
```
Comment (Mokto) [inline src/modules/profile/infos/name.tsx]: For readability, maybe it's better to move this function on top of the file ?

File: src/modules/profile/infos/phone.tsx [hunk 8]
```diff
@@ -29,6 +29,10 @@ const styles = StyleSheet.create({
 export const ProfileInfosPhone = observer(() => {
   const navigation = useNavigation();
   const profileFlowStore = useProfileFlowStore();
+  const userStore = useUserStore();
+
+  const [phone, SetPhone] = useState<string>(userStore?.profile?.phone.number || null);
```
Comment (Mokto) [inline src/modules/profile/infos/phone.tsx]: setPhone

File: src/modules/profile/infos/phone.tsx [hunk 9]
```diff
@@ -40,13 +44,30 @@ export const ProfileInfosPhone = observer(() => {
...
+        onChangeText={phone => SetPhone(phone)}
         keyboardType="number-pad"
       />
 
       <View style={styles.buttonContainer}>
-        <Button onPress={navigation.goBack}>Done</Button>
+        <Button
+          onPress={async () => {
+            if (userStore.profile._id) {
+              await userStore.patchProfile(userStore.profile._id, {
```
Comment (Mokto) [inline src/modules/profile/infos/phone.tsx]: I don't see why we await we the end of the result. Let's do it in "optimistic update" mode

File: src/router.tsx [hunk 10]
```diff
@@ -43,6 +44,7 @@ export const RouterMain = memo(() => {
       <Stack.Screen name="login-signup" component={LoginSignup} />
       <Stack.Screen name="profile-type" component={ProfileType} />
       <Stack.Screen name="profile-infos" component={InfosMain} />
+      <Stack.Screen name="Settings" component={authenticatedSettings} />
```
Comment (Mokto) [inline src/router.tsx]: Why is settings part of this ? It's supposed to be part of tabs

File: src/modules/profile/infos/phone.tsx [hunk 11]
```diff
@@ -40,13 +44,30 @@ export const ProfileInfosPhone = observer(() => {
...
-        onChangeText={(phone) => (profileFlowStore.phone = phone)}
+        value={phone}
+        onChangeText={phone => SetPhone(phone)}
         keyboardType="number-pad"
       />
 
       <View style={styles.buttonContainer}>
-        <Button onPress={navigation.goBack}>Done</Button>
+        <Button
+          onPress={async () => {
```
Comment (Mokto) [inline src/modules/profile/infos/phone.tsx]: Maybe move this function on top of the file ?

File: src/modules/authenticated/settings/index.tsx [hunk 12]
```diff
@@ -1,31 +1,68 @@
...
+        </View>
+
+        <View style={styles.buttonContainer}>
+          <CheckBoxButton
+            onPress={() => navigation.navigate('profile-infos-phone')}
+            checked={!!profileFlowStore.phone}
+          >
+            Phone
+          </CheckBoxButton>
+        </View>
```
Comment (Mokto) [inline src/modules/authenticated/settings/index.tsx]: If you are of `role=needed`, you should be able to edit your address, also

File: src/modules/profile/infos/phone.tsx [hunk 13]
```diff
@@ -29,6 +29,25 @@ const styles = StyleSheet.create({
 export const ProfileInfosPhone = observer(() => {
   const navigation = useNavigation();
   const profileFlowStore = useProfileFlowStore();
+  const userStore = useUserStore();
+
+  const [phone, setPhone] = useState<string>(
+    userStore?.profile?.phone?.number || null
```
Comment (mfateminia) [inline src/modules/profile/infos/phone.tsx]: shouldnt it be || '' ?

File: src/modules/authenticated/settings/index.tsx [hunk 14]
```diff
@@ -1,31 +1,85 @@
-import React, { memo } from 'react';
-
+import React, { useEffect } from 'react';
 import { useNavigation, CommonActions } from '@react-navigation/native';
-import { Button } from '../../../shared/button';
-import { useUserStore } from '../../../stores';
+import { ScrollView } from 'react-native-gesture-handler';
```
Comment (Mokto) [inline src/modules/authenticated/settings/index.tsx]: Import from react native

File: src/models/profile.ts [hunk 15]
```diff
@@ -22,3 +22,13 @@ export interface Profile {
   role: 'helper' | 'needer';
   phone: Phone;
 }
+
+export interface ProfilePatch {
```
Comment (Mokto) [inline src/models/profile.ts]: Can you use Profile from models ?

**Pull request merged**